File: backend/api/password_check.py
from flask import jsonify
import re

def check_password(password):
    # 1. Length Check
    if len(password) < 8:
        return jsonify({'msg': 'Password should be at least 8 characters!'}), 400
  
  # For upper or lower case
    if not re.search(r'[a-zA-Z]', password):
        return jsonify({'msg': 'Password must contain at least one letter (capital or small)! balances'}), 400
    
    # for num or special character
    if not re.search(r'[0-9!@#$%^&*(),.?":{}|<>]', password):
        return jsonify({'msg': 'Password must contain at least one number or special character!'}), 400
        
    return jsonify({'msg': 'Password Verified and accepted'}), 200

File: frontend/scripts/passwordCheck.js
import { showMessage } from './uiMessage.js';

export const passwordCheck = (password, msgBox) => {
  // 1. Length Check
  if (password.length < 8) {
    showMessage(msgBox, 'Password should be at least 8 characters.', 'error');
    return false;
  } 
  
  // 2. Check for at least one letter (Uppercase OR Lowercase)
  else if (!/[a-zA-Z]/.test(password)) {
    showMessage(msgBox, 'Password must contain at least one letter (A-Z or a-z).', 'error');
    return false;
  } 
  
  // 3. One number OR one special character
  else if (!/[0-9!@#$%^&*(),.?":{}|<>]/.test(password)) {
    showMessage(msgBox, 'Password must contain at least one number or special character.', 'error');
    return false;
  } 
  
  // All checks passed
  else {
    return true;
  }
};
